apache-rat-core/src/main/java/org/apache/rat/walker/DirectoryWalker.java

```diff
@@ -38,56 +38,42 @@ public class DirectoryWalker extends Walker implements IReportable {
 
     private static final FileNameComparator COMPARATOR = new FileNameComparator();
 
-    private final IOFileFilter directoryFilter;
-
-    /**
-     * Constructs a walker.
-     *
-     * @param file the directory to walk.
-     * @param directoryFilter directory filter to eventually exclude some directories/files from the scan.
-     */
-    public DirectoryWalker(File file, IOFileFilter directoryFilter) {
-        this(file, (FilenameFilter) null, directoryFilter);
-    }
+    private final IOFileFilter directoriesToIgnore;
 
     /**
      * Constructs a walker.
      *
      * @param file the directory to walk (not null).
-     * @param filter filters input files (optional),
+     * @param filesToIgnore filters input files (optional),
      *               or null when no filtering should be performed
-     * @param directoryFilter filters directories (optional), or null when no filtering should be performed.
+     * @param directoriesToIgnore filters directories (optional), or null when no filtering should be performed.
      */
-    public DirectoryWalker(File file, final FilenameFilter filter, IOFileFilter directoryFilter) {
-        super(file.getPath(), file, filter);
-        this.directoryFilter = directoryFilter;
+    public DirectoryWalker(File file, final FilenameFilter filesToIgnore, IOFileFilter directoriesToIgnore) {
+        super(file.getPath(), file, filesToIgnore);
```

Comment:
Maybe this class would also benefit from ScanDefault as mentioned in above comments